hkdfchacha20poly1305/hkdfchachapoly.go

```diff
@@ -0,0 +1,106 @@
+package hkdfchacha20poly1305
+
+// This package creates an AEAD using hkdf, chacha20, and poly1305
+// When sealing and opening, the hkdf is used to obtain the nonce and subkey for
+// chacha20. Other than the change for the how the subkey and nonce for chacha
+// are obtained, this is the same as chacha20poly1305
```

Comment:
Can you move this above the package? Ideally, have a look how godoc renders before and after the change.
(run `godoc -http=:6060` and open locally in your browser)
![image](https://user-images.githubusercontent.com/7441766/41633530-3c5c6366-73f4-11e8-8a0f-a7b54a4722db.png)